services/venice-server/src/main/java/com/linkedin/venice/stats/ServerHttpRequestStats.java

```diff
@@ -295,23 +295,35 @@ public ServerHttpRequestStats(
         () -> totalStats.earlyTerminatedEarlyRequestCountSensor,
         new OccurrenceRate());
 
-    if (isKeyValueProfilingEnabled) {
-      // size profiling is expensive
+    if (isKeyValueProfilingEnabled || requestType == RequestType.SINGLE_GET) {
+      final MeasurableStat[] valueSizeStats;
+      final MeasurableStat[] keySizeStats;
       String requestValueSizeSensorName = "request_value_size";
+      String requestKeySizeSensorName = "request_key_size";
+      if (isKeyValueProfilingEnabled) {
+        valueSizeStats = TehutiUtils
+            .getFineGrainedPercentileStatWithAvgAndMax(getName(), getFullMetricName(requestValueSizeSensorName));
+        keySizeStats = TehutiUtils
+            .getFineGrainedPercentileStatWithAvgAndMax(getName(), getFullMetricName(requestKeySizeSensorName));
+      } else {
+        valueSizeStats =
+            TehutiUtils.getPercentileStatWithAvgAndMax(getName(), getFullMetricName(requestValueSizeSensorName));
```

Comment:
Can we just use min, max, avg for sizes?

Reply:
Sorry, didn't see your comment and ended up force pushing